gdal_subsetter/exceptions.py

```diff
@@ -7,31 +7,42 @@
 """
 # noqa: D107
 
-from harmony_service_lib.exceptions import HarmonyException
+from harmony_service_lib.exceptions import HarmonyException, NoRetryException
 
 
-class HGAException(HarmonyException):
-    """Base class for exceptions in the Harmony GDAL Adapter."""
+class HGANoRetryException(NoRetryException):
+    """Base class for exceptions in the Harmony GDAL Adapter.
 
-    def __init__(self, message):
+    This exception is inherited by errors that should not cause Harmony to
+    retry the HGA step in a workflow chain, because it is known they will fail
+    upon that retry.
+
+    """
+
+    def __init__(self, message=None):
         super().__init__(message)
 
 
-class DownloadError(HGAException):
-    """Raised when the Harmony GDAL Adapter cannot retrieve input data."""
+class DownloadError(HarmonyException):
```

Comment:
I kind of wonder if the `HGANoRetryException` is an unnecessary intermediate layer. @flamingbear - you may have already said this when you realised that the service name wasn't actually being used as expected.

I didn't cut out the `HGANoRetryException` for now, but it feels like we could.